lib/ansible/modules/cloud/huawei/hwc_network_vpc.py

```diff
@@ -0,0 +1,511 @@
+        type: str
+    routes:
+        description:
+            - the route information.
+        returned: success
+        type: complex
+        contains:
+            destination:
+                description:
+                    - the destination network segment of a route.
+                returned: success
+                type: str
+            next_hop:
+                description:
+                    - the next hop of a route.
+                returned: success
+                type: str
+    enable_shared_snat:
+        description:
+            - show whether the shared snat is enabled.
+        returned: success
+        type: bool
+'''
+
+###############################################################################
+# Imports
+###############################################################################
+
+from ansible.module_utils.hwc_utils import (HwcSession, HwcModule,
+                                            DictComparison, navigate_hash,
+                                            remove_nones_from_dict,
+                                            remove_empty_from_dict,
+                                            are_dicts_different)
+import json
+import re
+import time
+
+###############################################################################
+# Main
+###############################################################################
+
+
+def main():
+    """Main function"""
+
+    module = HwcModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            name=dict(required=True, type='str'),
+            cidr=dict(required=True, type='str')
+        ),
+        supports_check_mode=True,
+    )
+    session = HwcSession(module, 'network')
+
+    state = module.params['state']
+
+    if (not module.params.get("id")) and module.params.get("name"):
+        module.params['id'] = get_id_by_name(session)
+
+    fetch = None
+    link = self_link(session)
+    # the link will include Nones if required format parameters are missed
+    if not re.search('/None/|/None$', link):
+        fetch = fetch_resource(session, link)
+        if fetch:
+            fetch = fetch.get('vpc')
+    changed = False
+
+    if fetch:
+        if state == 'present':
+            expect = _get_editable_properties(module)
+            current_state = response_to_hash(module, fetch)
+            if are_dicts_different(expect, current_state):
+                if not module.check_mode:
+                    fetch = update(session, self_link(session), [200])
+                    fetch = response_to_hash(module, fetch.get('vpc'))
+                    changed = True
+                else:
+                    changed = False
+            else:
+                fetch = current_state
+        else:
+            if not module.check_mode:
+                delete(session, self_link(session))
+                fetch = {}
+                changed = True
```

Comment:
The changed needs to be moved to the left as well.
```suggestion
            changed = True
```